### PR TITLE
Some fixes for skeleton patch related with merge & delete, undo/redo

### LINKED ISSUE #2157: Could not save annotations
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour
The annotations are saved without a problem

### Current Behaviour
A popup appears with the following message:

Could not save annotations
Error: Request failed with status code 400. "duplicate key value violates unique constraint \"engine_trackedshape_pkey\"\nDETAIL: Key (id)=(113) already exists.\n".

### Steps to Reproduce (for bugs)

I am annotating a video with the tracking bounding boxes. When trying to save mid way through, the error appears.

### Context
I am not able to save any annotations. All my work will be lost, when I close the window. When opening the job in a 2nd window it show the last save was 400 frames ago.

Edit: I was able to "save" my work by downloading the annotation via the "Menu" -> "Dump annotations" / "Export Dataset" Buttons.

### Your Environment

- Git hash commit (`git log -1`): cb4d8e6ee571594f74ff96bc6cbc7c597c732e4a
- Docker version `docker version` (e.g. Docker 17.0.05): 19.03.5
- Are you using Docker Swarm or Kubernetes? no
- Operating System and version (e.g. Linux, Windows, MacOS): Ubuntu Desktop
- Code example or link to GitHub repo or gist to reproduce problem:
- Other diagnostic information / logs:
    <details>
    <summary>
2020-09-09 07:05:44,580 DEBG 'runserver' stderr output:
[Wed Sep 09 07:05:44.580126 2020] [wsgi:error] [pid 630:tid 139984631072512] [remote 172.28.0.7:54352] WARNING:django.request:Bad Request: /api/v1/jobs/2/annotations
</summary>
    </details>

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
Resolved #2157

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by bsekachev** [comment]
@klakhov Could you please look at this patch carefully and try it?

**Review by klakhov — COMMENTED** [review]
Btw what do you think about making merge selection for sceletons more noticable like its done with box tracks?
image
image

**Comment by bsekachev** [comment]
> Btw what do you think about making merge selection for sceletons more noticable like its done with box tracks?

@klakhov 

I do not think it will work for skeletons, because they are not ``closed`` shapes. 
We can only fill skeleton points to blue line like it is done for ``points`` shape.

File: cvat-core/src/annotations-objects.ts [hunk 1]
```diff
@@ -495,6 +501,15 @@ class Annotation {
...
         throw new ScriptingError('Is not implemented');
     }
+
+    public get removed(): boolean {
+        return this._removed;
+    }
+
+    public set removed(value: boolean) {
+        this._clearServerID();
+        this._removed = value;
```
[the remark this answers is not in the document]
Comment (bsekachev) [reply cvat-core/src/annotations-objects.ts]: Done

File: cvat-core/src/annotations-collection.ts [hunk 2]
```diff
@@ -351,18 +378,18 @@
             this.history.do(
                 HistoryActions.MERGED_OBJECTS,
                 () => {
-                    trackModel.removed = true;
+                    imported.tracks[0].removed = true;
```
[the remark this answers is not in the document]
Comment (bsekachev) [reply cvat-core/src/annotations-collection.ts]: Done

**Comment by bsekachev** [comment]
> We can only fill skeleton points to blue line like it is done for points shape.

@klakhov 

I did so.

**Review by klakhov — APPROVED** [review]
Good job!

**Pull request merged**